src/libstore-tests/http-binary-cache-store.cc

```diff
@@ -34,4 +38,86 @@ TEST(HttpBinaryCacheStore, constructConfigWithParamsAndUrlWithParams)
     EXPECT_EQ(config.getReference().params, params);
 }
 
+struct ServerManager
+{
+    httplib::Server & server;
+    std::thread & server_thread;
+
+    ~ServerManager()
+    {
+        server.stop();
+        if (server_thread.joinable()) {
+            server_thread.join();
+        }
+    }
+};
+
+TEST(HttpBinaryCacheStore, NixCacheInfoBasicTest)
+{
+// FIXME: MacOS tests on Github actions were not able to bind to any port,
+// it kept returning -1 for the bound port.
+#if !defined(__linux__)
+    GTEST_SKIP() << "Skipping test on non-Linux platform due to unknown networking differences.";
+#endif
```

Comment:
@Ericson2314 I couldn't figure out why but the MacOS tests were not letting me opening up a port.
I tried "localhost" if it was IPV6 only but that also failed...

Reply:
I think we need to add `__darwinAllowLocalNetworking`
Might be worth adding irrespective of the change to match Linux.